### ISSUE TITLE
Core on Test: Merging Core date ranges doesn't work for multiple ranges

### ISSUE THREAD

**Issue by ostephens** [k-int/KBPlus#907]
If there is an existing Core date range of 2014-01-01 to 2015-12-31 and a new date range of 2013-01-01 to 2013-12-31 is added, these are merged to give a single range of 2013-01-01 to 2015-12-31

However if there are multiple, non-overlapping, Core date ranges already existing, then any added Core date ranges will only be merged with one of the dates, not with all of them.

e.g. Two Core date ranges:
2014-01-01 to 2015-12-31
2012-01-01 to 2012-12-31

Adding a new date range of

2013-01-01 to 2013-12-31
or
2010-01-01 to 2015-12-31

still results in two date ranges (one of the existing date ranges will be extended, and the other will stay as it is).

All overlapping/adjacent date ranges should be merged when the 'apply' button is clicked.

This is probably a bit of an edge case so marked as low priority

**Comment by iyawnis** [comment]
Yes I believe we have talked with Ian about this before, and it is a bit complicated to solve ( although possible - possibly ) will add to back log
